### ISSUE TITLE
connection string port ignored - issue

### ISSUE THREAD

**Issue by mohamedhagag** [graphile/postgraphile#1494]
when using the command --> port is 5414
```
postgres://mhagag:****@127.0.0.1:5414/atst1
```

The port is always ignored and I get the below error --> trying to connect the default port 5432:
```
Error: connect ECONNREFUSED 127.0.0.1:5432
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16)
```
Which means that the command ignores the specified port in the connection string, below is the full output:
```
PostGraphile v4.12.2 server listening on port 5000 🚀

  ‣ GraphQL API:         http://localhost:5000/graphql
  ‣ GraphiQL GUI/IDE:    http://localhost:5000/graphiql (RECOMMENDATION: add '--enhance-graphiql')
  ‣ Postgres connection: postgres://mhagag:[SECRET]@127.0.0.1/atst1
  ‣ Postgres schema(s):  public
  ‣ Documentation:       https://graphile.org/postgraphile/introduction/
  ‣ Node.js version:     v14.15.1 on linux x64
  ‣ Join Sterblue in supporting PostGraphile development: https://graphile.org/sponsor/

* * *

A serious error occurred when building the initial schema. Exiting because `retryOnInitFail` is not set. Error details:

Error: connect ECONNREFUSED 127.0.0.1:5432
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16)
```

**Note**: psql postgres://mhagag:****@127.0.0.1:5414/atst1 works fine with same URI

Appreciate your kind support @benjie

**Comment by benjie** [comment]
Can confirm this affects v4.12 but not v4.11. Seems to be that `pg-connection-string`'s parse method now returns the port as a string.

**Comment by benjie** [comment]
Fix released in 4.12.3
